fabric_rti_mcp/server.py

```diff
@@ -1,34 +1,166 @@
+    
+    parser = argparse.ArgumentParser(description="Fabric RTI MCP Server")
+    transport_group = parser.add_mutually_exclusive_group()
+    transport_group.add_argument("--stdio", action="store_true", help="Use stdio transport (default)")
+    transport_group.add_argument("--http", action="store_true", help="Use HTTP transport")
+    
+    parser.add_argument("--host", default=env_host,help="Host to bind to when running in HTTP mode")
+    parser.add_argument("--port", type=int, default=env_port, help="Port to use when running in HTTP mode")
+    parser.add_argument("--streamable-http-path", default=env_path, help="Path for HTTP transport")
+    
+    args = parser.parse_args()
+    
+    # Determine transport mode from arguments
+    # for Azure (PORT env variable is set)
+    if args.http or os.getenv("PORT"):
+        mode = "http"
+    elif args.stdio:
+        mode = "stdio"
+    else:
+        mode = "stdio"  # Default
+    
+    return {
+        "mode": mode,
+        "host": args.host,
+        "port": args.port,
+        "streamable_http_path": getattr(args, 'streamable_http_path')
+    }
+
+
 def register_tools(mcp: FastMCP) -> None:
+    """Register all tools with the MCP server."""
     logger.info("Kusto configuration keys found in environment:")
     logger.info(", ".join(kusto_config.KustoConfig.existing_env_vars()))
+    
     kusto_tools.register_tools(mcp)
     eventstream_tools.register_tools(mcp)
+    
+from datetime import datetime
 
+# Health check function defined at module level
+async def health_check(request: Request) -> JSONResponse:
+    """Health check endpoint."""
+    return JSONResponse({
+        "status": "healthy",
+        "timestamp": datetime.now().isoformat(),
+        "server": "fabric-rti-mcp"
+    })
 
+def add_health_endpoint(mcp: FastMCP) -> None:
+    """Add health endpoint for Kubernetes liveness probes."""
+    # Register the pre-defined health check function
+    mcp.custom_route("/health", methods=["GET"])(health_check)
+    
 def main() -> None:
-    # writing to stderr because stdout is used for the transport
-    # and we want to see the logs in the console
-    logger.error("Starting Fabric RTI MCP server")
-    logger.error(f"Version: {__version__}")
-    logger.error(f"Python version: {sys.version}")
-    logger.error(f"Platform: {sys.platform}")
-    # print pid
-    logger.error(f"PID: {os.getpid()}")
-    # import later to allow for environment variables to be set from command line
-    mcp = FastMCP("fabric-rti-mcp-server")
-    register_tools(mcp)
-    mcp.run(transport="stdio")
+    """Main entry point for the server."""
+    try:
+        # Set up signal handlers for graceful shutdown
+        signal.signal(signal.SIGINT, setup_shutdown_handler) # Signal Interrupt)
+        signal.signal(signal.SIGTERM, setup_shutdown_handler) # Signal Terminate
+        
+        # Parse command line arguments
+        args = parse_args()
```

Comment:
use the config classes.. they have something like `ConfigClass.from_env()` that you can extend